File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    score = {'R':0, 'T':0, 'C':0, 'F':0, 'J':0, 'M':0, 'A':0, 'N':0}
    
    for s, c in zip(survey, choices):
        if c in [1, 2, 3]:
            score[s[0]] += (4-c)
        elif c in [5, 6, 7]:
            score[s[1]] += (c-4)
    
    print(score)
    
    answer = ''
    
    if score['R'] >= score['T']:
        answer += 'R'
    else:
        answer += 'T'
    if score['C'] >= score['F']:
        answer += 'C'
    else:
        answer += 'F'
    if score['J'] >= score['M']:
        answer += 'J'
    else:
        answer += 'M'
    if score['A'] >= score['N']:
        answer += 'A'
    else:
        answer += 'N'
        
    return answer

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.swift
import Foundation

func solution(_ survey:[String], _ choices:[Int]) -> String {
    var testResult: [Character: Int] = [
        "R": 0, "T": 0, "C": 0, "F": 0,
        "J": 0, "M": 0, "A": 0, "N": 0
    ]

    for (s, c) in zip(survey, choices) {
        let point = getPoint(of: c)
        if c < 4 {  // 비동의 - 앞 캐릭터
            testResult[s.first!, default: 0] += point
        } else if c > 4 {  // 동의 - 뒤 캐릭터
            testResult[s.last!, default: 0] += point
        }
    }
    
    return getResult(of: testResult)
}

func getPoint(of choice: Int) -> Int {
    return choice <= 4 ? 4 - choice : choice - 4
}

func getResult(of testResult: [Character: Int]) -> String {
    var result = ""
    print(testResult)
    result += testResult["R"]! >= testResult["T"]! ? "R" : "T"
    result += testResult["C"]! >= testResult["F"]! ? "C" : "F"
    result += testResult["J"]! >= testResult["M"]! ? "J" : "M"
    result += testResult["A"]! >= testResult["N"]! ? "A" : "N"
    
    return result
}
